WordPress/Classes/Services/MediaCoordinator.swift

```diff
@@ -330,13 +361,13 @@ class MediaCoordinator: NSObject {
             self.fail(nserror, media: media)
         }
 
-        coreDataStack.performAndSave { context in
-            let service = self.mediaServiceFactory.create(context)
-            var progress: Progress? = nil
-            service.uploadMedia(media, automatedRetry: automatedRetry, progress: &progress, success: success, failure: failure)
-            if let progress {
-                resultProgress.addChild(progress, withPendingUnitCount: resultProgress.totalUnitCount)
-            }
+        // For some reason, this `MediaService` instance has to be created with the main context, otherwise
+        // the successfully uploaded media is shown as a "local" assets incorrectly.
+        let service = self.mediaServiceFactory.create(coreDataStack.mainContext)
```

Comment:
I think this change—using the main context—fixes [the issue mentioned in this comment](https://github.com/wordpress-mobile/WordPress-iOS/issues/20298#issuecomment-1465319707).

Reply:
In such case, what do you think of adding a link to the issue?

```suggestion
        // 
        // See https://github.com/wordpress-mobile/WordPress-iOS/issues/20298
        let service = self.mediaServiceFactory.create(coreDataStack.mainContext)
```

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/20322/commits/65b7249674e554113fb4e6b983d8ff4997660d02